src/OrchardCore.Modules/OrchardCore.Liquid/ScriptsMiddleware.cs

```diff
@@ -28,33 +31,44 @@ public async Task Invoke(HttpContext httpContext)
         {
             if (httpContext.Request.Path.StartsWithSegments("/OrchardCore.Liquid/Scripts", StringComparison.OrdinalIgnoreCase))
             {
-                var script = default(string);
                 if (Path.GetFileName(httpContext.Request.Path.Value) == "liquid-intellisense.js")
                 {
-                    var cache = httpContext.RequestServices.GetRequiredService<IMemoryCache>();
-                    const string key = "OrchardCore.Liquid/Scripts/liquid-intellisense.js";
-                    if (!cache.TryGetValue(key, out script))
+                    if (httpContext.Request.Headers.TryGetValue(HeaderNames.IfNoneMatch, out var v))
                     {
+                        if (v.Contains(etag))
+                        {
+                            httpContext.Response.StatusCode = StatusCodes.Status304NotModified;
+                            return;
+                        }
+                    }
 
+                    var cacheControl = "public, max-age=2592000, s-max-age=31557600";
```

Comment:
nit: static value. Could use `$"{Timespan.FromHours().TotalSeconds}"` to make it obvious how much that is.

Reply:
done.